appveyor/numpy.patch

```diff
@@ -0,0 +1,831 @@
+                 if libname in clibs and clibs[libname] != build_info:
+-                    log.warn('library %r defined more than once,'\
+-                             ' overwriting build_info\n%s... \nwith\n%s...' \
++                    log.warn('library %r defined more than once,'
++                             ' overwriting build_info\n%s... \nwith\n%s...'
+                              % (libname, repr(clibs[libname])[:300], repr(build_info)[:300]))
+                 clibs[libname] = build_info
+         # .. and distribution libraries:
+@@ -181,7 +190,7 @@ class build_ext (old_build_ext):
+             elif 'f77' in ext_languages:
+                 ext_language = 'f77'
+             else:
+-                ext_language = 'c' # default
++                ext_language = 'c'  # default
+             if l and l != ext_language and ext.language:
+                 log.warn('resetting extension %r language from %r to %r.' %
+                          (ext.name, l, ext_language))
+@@ -196,9 +205,9 @@ class build_ext (old_build_ext):
+         # Initialize C++ compiler:
+         if need_cxx_compiler:
+             self._cxx_compiler = new_compiler(compiler=compiler_type,
+-                                             verbose=self.verbose,
+-                                             dry_run=self.dry_run,
+-                                             force=self.force)
++                                              verbose=self.verbose,
++                                              dry_run=self.dry_run,
++                                              force=self.force)
+             compiler = self._cxx_compiler
+             compiler.customize(self.distribution, need_cxx=need_cxx_compiler)
+             compiler.customize_cmd(self)
+@@ -238,7 +247,7 @@ class build_ext (old_build_ext):
+                                                dry_run=self.dry_run,
+                                                force=self.force,
+                                                requiref90=True,
+-                                               c_compiler = self.compiler)
++                                               c_compiler=self.compiler)
+             fcompiler = self._f90_compiler
+             if fcompiler:
+                 ctype = fcompiler.compiler_type
+@@ -256,6 +265,36 @@ class build_ext (old_build_ext):
+         # Build extensions
+         self.build_extensions()
+ 
++        mingw_bin = os.path.dirname(self.which('gfortran.exe'))
++        # Copy all DLLs to the correct folder
++        dlls = glob(os.path.join(self.build_temp, '*.dll')) + \
++            glob(os.path.join(self.build_clib, '*.dll')) + \
++            glob(os.path.join(mingw_bin, '*.dll')) + \
++            self.dlls
++        dlls = list(set(dlls))
++
++        dll_folder = os.path.join(self.build_lib, 'scipy', '_lib')
++        with suppress(OSError):
++            os.makedirs(dll_folder)
++        for dll in dlls:
++            shutil.copy(dll, dll_folder)
++
```

Comment:
We need to copy all of the DLLs that we have built plus the mingw DLLs plus the OpenBLAS DLL so that our extension can access it.